target/hexagon/gdbstub.c

```diff
@@ -39,6 +39,12 @@ int hexagon_gdb_read_register(CPUState *cs, GByteArray *mem_buf, int n)
 
     n -= TOTAL_PER_THREAD_REGS;
 
+    if (n < NUM_PREGS) {
+        return gdb_get_reg8(mem_buf, env->pred[n]);
+    }
+
+    n -= NUM_PREGS;
```

Comment:
I line 46 needed?

Reply:
I think it's a pattern for the sake of someone who might add a new block of registers to follow.  But in practice, no, probably not.